### ISSUE TITLE
Multipole expansions

### ISSUE THREAD

**Issue by mjyb16** [Ciela-Institute/caustics#139]
Is there a plan for adding multipole expansion parameters to lenses? I don't see any mentions of multipole capability at the moment in the codebase or the docs.

**Comment by ConnorStoneAstro** [comment]
Hmmm, could you describe more of what you are looking for here? Is there a reference where this is done? For a lot of lenses they are quite simple to evaluate and a multipole expansion would probably just increase the computational graph. An exception might be the `NFW` and `TNFW` lenses which are a bit of a pain right now. Or are you thinking of a new lens class that is just a multipole expansion up to a certain order and can take arbitrary values?

**Comment by AlexandreAdam** [comment]
Multipole should be an additional lens class. The external convergence is a very special case of a multipole extension, since it is the zeroth moment of the multipole expansion of the external convergence. 

A good reference on the subject is [Schneider & Bartelmann (1997)](https://articles.adsabs.harvard.edu/pdf/1997MNRAS.286..696S), but one can find also a more modern treatment in [Bernstein & Nakajima](https://iopscience.iop.org/article/10.1088/0004-637X/693/2/1508/pdf). 

Note that Yashar's [SDP.81 paper](https://arxiv.org/abs/1601.01388) found that multiples were very important in the detection of substructure. He gives a simple definition of the multiple expansion of the convergence in polar coordinates
```math
\kappa(r, \theta) = \left(\frac{r}{r_s}\right)^\alpha [A_m \cos (m \phi) + B_m \sin(m \phi)]
```
In his paper he describes that the logarithmic slope $\alpha$ is shared with his main lens parameters and he chooses $r_s$ based on the size of his lens (?).

**Comment by ConnorStoneAstro** [comment]
closed by #215
